### PR TITLE
test(modal): migrate tests to Playwright

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

This PR migrates existing modal E2E tests from Puppeteer to Playwright. A few notable changes:

1. I added the Playwright equivalent of `dragElementBy`. Let me know if you'd prefer that I pull this out into its own PR.
2. Any explicit RTL tests have been removed since our Playwright setup automatically runs RTL tests.
3. I removed the additional card tests that tested the rendering of it on different viewports in favor of one test. Our Playwright setup automatically runs tests in mobile vs desktop viewports.
4. Any test that took a screenshot of the modal only when it was presented was consolidated into a single "visual regression" test.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by sean-perkins — APPROVED** [review]

File: core/src/components/modal/test/sheet/modal.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    await page.click('#backdrop-active');
+
+    await ionModalDidPresent.next();
+
+    await page.mouse.click(50, 50);
+
+    await ionModalDidDismiss.next();
+  });
+  test('should present another the sheet modal when clicking an inactive backdrop', async ({ page }) => {
```
Comment (amandaejohnston) [inline core/src/components/modal/test/sheet/modal.e2e.ts:33]:
```suggestion
  test('should present another sheet modal when clicking an inactive backdrop', async ({ page }) => {
```

File: core/src/components/modal/test/sheet/modal.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    await page.click('#backdrop-inactive');
+    await ionModalDidPresent.next();
+
+    await page.click('#custom-height-modal');
+    await ionModalDidPresent.next();
+
+    await page.waitForSelector('.custom-height', { state: 'visible' });
+  });
+  test('input should outside sheet modal should be focusable when backdrop is inactive', async ({ page }) => {
```
Comment (amandaejohnston) [inline core/src/components/modal/test/sheet/modal.e2e.ts:44]:
```suggestion
  test('input outside sheet modal should be focusable when backdrop is inactive', async ({ page }) => {
```

Comment (amandaejohnston) [inline core/src/components/modal/test/sheet/modal.e2e.ts:44]: Looks like the test `'input should not be focusable when backdrop is active'` was skipped, was that intentional?

File: core/src/components/modal/test/basic/modal.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('modal: focus trapping', () => {
```
Comment (amandaejohnston) [inline core/src/components/modal/test/basic/modal.e2e.ts:4]: Now that we're manually opening/checking all modals, it looks like `openModal` and `testModal` in `test.utils.ts` are no longer needed. Could we remove that file as well?

**Review by amandaejohnston — CHANGES_REQUESTED** [review]
Great work here! Mostly minor things below.

File: core/src/components/modal/test/sheet/modal.e2e.ts (hunk 2 above)
Comment (liamdebeasi) [reply core/src/components/modal/test/sheet/modal.e2e.ts]: Just added the test -good catch!

File: core/src/components/modal/test/basic/modal.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    await ionModalDidPresent.next();
+
+    /**
+     * The default WebKit behavior is to
+     * highlight items on webpages with Option-Tab.
+     * See "Press Tab to highlight each item on a webpage"
+     * in Safari Preferences > Advanced.
+     */
+    if (browserName === 'webkit') {
+      await page.keyboard.down('Alt');
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply core/src/components/modal/test/basic/modal.e2e.ts:20]: The Alt key should be released. Will fix

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**